src/main/java/com/uwetrottmann/tmdb2/TmdbInterceptor.java

```diff
@@ -39,10 +70,95 @@ public static Response handleIntercept(Chain chain, String apiKey) throws IOExce
         HttpUrl.Builder urlBuilder = request.url().newBuilder();
         urlBuilder.setEncodedQueryParameter(Tmdb.PARAM_API_KEY, apiKey);
 
+        if (request.url().pathSegments().get(0).equals("account")) {
```

Comment:
A user can have Guest and Account Session simultaneously, which means he is able to rate a movie with his account, and one as a guest. And you will never know what a user exactly really things, and if he wants to do this, I believe beside that you proposed, which is fine, To do the following,

Add a parameter on each conflict method, called authentication, with this type:

```
public enum AuthenticationType {
    ACCOUNT("account"),
    GUEST("guest");

    private static final Map<String, AuthenticationType> lookup = prepareLookup();

    private static Map<String,AuthenticationType> prepareLookup() {
        Map<String,AuthenticationType> atMap = new HashMap<>();
        for (AuthenticationType lang : AuthenticationType.values()) {
            atMap.put(lang.value, lang);
        }
        return atMap;
    }

    private final String value;

    public static AuthenticationType get(String value) {
        return lookup.get(value);
    }

    AuthenticationType(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return value;
    }
}
```
Then determine by the interceptor what user really wants to do, and leave the rest to the authenticator, In any case i believe, when a user calls the getAccountSession, or getGuestSession, we should try to authenticate them immediately instead of having an extra request with the authenticator for no reason at all, and leave the authenticator handle the expired sessions in any case.